jax/lax/lax_control_flow.py

```diff
@@ -389,12 +389,86 @@ def _while_loop_jvp(primals, tangents, cond_nconsts, cond_jaxpr, body_nconsts,
                   for nz in nonzeros_out]
   return out_carry, out_tangents
 
+def _while_partial_eval(trace, *tracers: pe.Tracer, cond_nconsts: int,
```

Comment:
This should not be here, my mistake. This is part of a follow-up work to fix #2129. It will come it as a separate PR.

Reply:
I have forced push an update that removes the while_partial_eval